app/src/main/java/com/example/ncov19traking/ui/global/GlobalFragment.kt

```diff
@@ -94,6 +106,37 @@ class GlobalFragment : Fragment() {
         })
     }
 
+    private fun setupPieChart(pieChart: PieChart) {
+        pieChart.apply {
+            animateY(1500)
+            setDrawEntryLabels(false)
+            setUsePercentValues(false)
+            setHoleColor(ContextCompat.getColor(context, R.color.backgroundColor))
+            description.isEnabled = false
+            legend.isEnabled = false
+        }
+    }
+
+    private fun getChartData(nCoVInfo: NCoVInfo): PieData {
+        val pieEntry = ArrayList<PieEntry>()
+        pieEntry.add(PieEntry(nCoVInfo.cases.toFloat()))
+        pieEntry.add(PieEntry(nCoVInfo.recovered.toFloat()))
+        pieEntry.add(PieEntry(nCoVInfo.deaths.toFloat()))
+        return PieData(PieDataSet(pieEntry, getString(R.string.piechart_dataset_label)).apply {
+            colors = setChartColors()
+            setDrawValues(false)
+            sliceSpace = 2F
+        })
+    }
+
+    private fun setChartColors(): List<Int> {
+        val colors = ArrayList<Int>(3)
+        colors.add(ContextCompat.getColor(requireContext(), R.color.textColor))
+        colors.add(ContextCompat.getColor(requireContext(), R.color.recoveredColor))
+        colors.add(ContextCompat.getColor(requireContext(), R.color.deathsColor))
+        return colors
+    }
```

Comment:
This is not that important I will leave the comment just so you take this into account. All this methods control the chart config, maybe they could be moved to a class in charge of configuring all the pie chart specific code and configurations, this would clean up the fragment and all the responsability would be moved to this new class.